drivers/adc/adc_nxp_s32_adc_sar.c

```diff
@@ -0,0 +1,458 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/irq.h>
+#include <Adc_Sar_Ip_HwAccess.h>
+#include <Adc_Sar_Ip.h>
+#include <Adc_Sar_Ip_Irq.h>
+
+#define ADC_CONTEXT_USES_KERNEL_TIMER
+#include "adc_context.h"
+
+#define DT_DRV_COMPAT nxp_s32_adc_sar
+LOG_MODULE_REGISTER(adc_nxp_s32_adc_sar, CONFIG_ADC_LOG_LEVEL);
+
+/* ADC internal reference voltage in mV */
+#define ADC_NXP_S32_INTERNAL_VREF 3300
```

Comment:
should this be a DT property? it might be more scalable in case this changes for another platform or version of IP

Reply:
I agree. I updated to use  DT prop for vref.